src/main/java/com/ontology/service/impl/ToolsServiceImpl.java

```diff
@@ -0,0 +1,139 @@
+    @Autowired
+    private OrderDataMapper orderDataMapper;
+
+    @Override
+    public List<OrderListResp> queryList(String action, Integer provider, String buyerOntid) {
+        String queryType = null;
+        if (0==provider) {
+            // 需求方
+            queryType = "buyer_ontid";
+        } else if (1==provider) {
+            // 提供方
+            queryType = "seller_ontid";
+        } else {
+            throw new OntIdException(action, ErrorInfo.PARAM_ERROR.descCN(), ErrorInfo.PARAM_ERROR.descEN(), ErrorInfo.PARAM_ERROR.code());
+        }
+        List<Order> orderList = orderMapper.getOrderList(queryType,buyerOntid);
+        List<OrderListResp> resps = new ArrayList<>();
+        for (Order order:orderList) {
+            OrderListResp resp = new OrderListResp();
+            resp.setOrderId(order.getOrderId());
+            resp.setBuyDate(order.getBuyDate());
+            if (0==provider) {
+                resp.setDataProvider(order.getSellerOntid());
+            } else {
+                resp.setDataDemander(order.getBuyerOntid());
+            }
+            resp.setState(order.getState());
+            List<String> dataList = new ArrayList<>();
+            for (OrderData data:order.getOrderData()){
+                String dataId = data.getDataId();
+                dataList.add(dataId);
+            }
+            resp.setDataIdList(dataList);
+            resps.add(resp);
+        }
+        return resps;
+    }
+
+    @Override
+    public String encrypt(String action, String ddo, Integer kid, String message) {
+        String publicKey = getPublicKey(action,ddo,kid);
+
+        String secStr = JSON.toJSONString(ECIES.Encrypt(publicKey,message.getBytes()));
+        return secStr;
+    }
+
+    private String getPublicKey(String action, String ddo, Integer kid) {
+        String publicKey = null;
+        JSONObject jsonObject = JSONObject.parseObject(ddo);
+        JSONArray owners = jsonObject.getJSONArray("Owners");
+        for (int i=0;i<owners.size();i++) {
+            JSONObject obj = owners.getJSONObject(i);
+            if (obj.getString("PubKeyId").endsWith("keys-"+kid)) {
+                publicKey = obj.getString("Value");
+                break;
+            }
+        }
+        if (Helper.isEmptyOrNull(publicKey)) {
+            throw new OntIdException(action, ErrorInfo.NOT_EXIST.descCN(), ErrorInfo.NOT_EXIST.descEN(), ErrorInfo.NOT_EXIST.code());
+        }
+        return publicKey;
+    }
+
+    @Override
+    public String decrypt(String action, String ontid, String ddo, Integer kid, String password, String secStr) throws Exception {
+        String publicKey = getPublicKey(action,ddo,kid);
+
+        OntId OntId = getOntId(action, ontid, password);
+        JSONObject jsonObject = JSONObject.parseObject(OntId.getKeystore());
+        if (publicKey.equals(jsonObject.getString("publicKey"))) {
+            Account buyerAcct = sdk.getAccount(OntId.getKeystore(), password);
```

Comment:
Maybe we should change it's name to `acct`. In general, anyone with `ONT ID` can use this interface.

```java
Account acct = sdk.getAccount(OntId.getKeystore(), password);
```

Reply:
ok